File: Api/Models/PostModel.cs
using System;
using System.Collections.Generic;

namespace Affix.Models
{
    public class PostModel
    {
        public string Title { get; set; }

        public string Content { get; set; }

        public string Summary { get; set; }

        public string Header { get; set; }

        public string Moniker { get; set; }

        public DateTime Date { get; set; }

        public int Likes { get; set; }

        public int Shares { get; set; }

        public Guid ImageId { get; set; }

        public string ImageAltText { get; set; }

        public bool IsDraft { get; set; }

        public string Category { get; set; }

        public string Tags { get; set; }

        public override bool Equals(object obj)
        {
            var other = obj as PostModel;

            if (other == null)
            {
                return false;
            }

            return Moniker.ToLowerInvariant() == other.Moniker.ToLowerInvariant();
        }

        public override string ToString()
        {
            return $"Title: {Title}\n" +
                $"Content: {Content}\n" +
                $"Summary: {Summary}\n" +
                $"Header: {Header}\n" +
                $"Moniker: {Moniker}\n" +
                $"Date: {Date}\n" +
                $"Likes: {Likes}\n" +
                $"Shares: {Shares}\n" +
                $"ImageId: {ImageId}\n" +
                $"ImageAltText: {ImageAltText}\n" +
                $"IsDraft: {IsDraft}\n" +
                $"Category: {Category}\n" +
                $"Tags: {Tags}";

        }
    }
}

File: Client/projects/affix-client/src/app/domain/post/models/post.model.ts
export class PostModel {
    public title: string;
    public date: Date;
    public moniker: string;
    public imageId: string;
    public imageAltText: string;
    public isDraft: boolean;
    // card properties
    public header: string;
    public summary: string;
    public imageSrc: string;
    public index: number;
    // details properties
    public content: string;
    public parsedContent: string;
    public category: string;
    public tags: string;

constructor(
    title: string,
    date: Date,
    moniker: string,
    imageId: string,
    imageAltText: string,
    isDraft: boolean,
    header: string,
    summary: string,
    imageSrc: string,
    index: number,
    content: string,
    parsedContent: string,
    category: string,
    tags: string) {
        this.title = title;
        this.date = date;
        this.moniker = moniker;
        this.imageId = imageId;
        this.imageAltText = imageAltText;
        this.isDraft = isDraft;
        this.header = header;
        this.summary = summary;
        this.imageSrc = imageSrc;
        this.index = index;
        this.content = content;
        this.parsedContent = parsedContent;
        this.category = category;
        this.tags = tags;
    }
}